CSharpFunctionalExtensions/Result/Extensions/With/ResultOfT.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+
+namespace CSharpFunctionalExtensions
+{
+    public static partial class ResultExtensions
+    {
+        public static Result<(T1, T2)> With<T1, T2>(this Result<T1> a, Result<T2> b)
+        {
+            return a.WithMap(b, (x, y) => (x, y));
```

Comment:
Use `static () {}` lambda in all instances where no closure allocation is required.